### ISSUE TITLE
[Accessibility] Support character-tracking in repl

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#195372]
Type: <b>Feature Request</b>

I understand that command history is not supported in an external repl outside the shell integration. It would be so nice if this could be supported in the future for screen reader users.

1. Execute `Python: Start REPL` in command palette.

1. In the Python repl, type `import os` and hit Enter.

1. Hit UpArrow.

1. Hit LeftArrow.

* Notes: Screen reader does not announce the command history.

VS Code version: Code - Insiders 1.84.0-insider (6a7a661757dec1983ff05ef908a2bbb75ce841e0, 2023-10-11T05:34:23.077Z)
OS version: Windows_NT x64 10.0.22621
Modes:

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by Tyriar** [comment]
FYI @anthonykim1

**Comment by rohanshardha** [comment]
This is an accessibility issue and compared to sighted coders not many people are there who use screen reader to code. I think this is unfair to require 20 upvotes for an accessibility issue. Should take into account the propotions.

**Comment by Tyriar** [comment]
@rohanshardha shell integration is actually coming to the python extension's start repl command very soon. Without shell integration there's simply no way to reliably do what you ask as the terminal is just a black box of text when not using some sort of shell integration telling us where things are. Actually, one of the major drivers for shell integration has always been to improve accessibility.

For terminals without shell integration we now have the "accessible view" which allows you to navigate the terminal buffer with an editor similar to the main editor. It's less convenient but I believe you can always use this when shell integration isn't working.

Closing since this is essentially done for the highlighted use case and I'm not sure what other actions we could do to improve it generally.
